scripts/zones/Windurst_Walls/Zone.lua

```diff
@@ -41,6 +42,11 @@ function onZoneIn(player, prevZone)
     return cs
 end
 
+function afterZoneIn(player)
```

Comment:
@zach2good  having issues adding it inside OnGameIn as player:messageSpecial cannot be shown until afterZoneIn

Tried adding afterZoneIn inside player.lua but hit a stump since i think it relies on being used only inside the zone.lua files inside the zones folder

Reply:
Might need to fetch the ID file for the player's current zone

Reply:
Hi @ibm2431  :cat: 

All the logic works great (points being added only once a day) onGameIn but any messages sent to the player won't show because the client isn't ready to display chat packets yet.

Reply:
There is a packet timing issue, if the client gets the message before it thinks it is doesn't loading the zone (on the clients side) it will drop the message without placing it into the chat log. a similar issue exists with the change music packet. If that is delayed bay some amount either by other things taking time to execute before the server sends, or an intentional delay (via the evil `timer()` function) it will display just fine in onGameIn. ***This exact issue*** is why I was desiring an afterGameIn() to be added to player.lua awhile back. `timer()` is a buggy sob in many situations but would be less evil than doing this per zone.

Please note that the timing is not consistent and can vary based on what the client PC -or- the server were doing. Can be as short as 200ms on fast hardware with minimal traffic, or as long as 2000ms (2 full seconds). And not that timer does not like being handed any function that wasn't a binding so you need do all your logic then call the timer just for the message, or face potential errors (unless you wanna rewrite `timer()` too).

```
player:timer(5000, function (player) player:messageSpecial(ID.text.CARRIED_OVER_POINTS, 0, 1500) end)
```

Even tho I am demonstrating it, I despise using timer. Up to staff how they wanna handle the situation.

Reply:
Indeed, this timing issue is just a symptom of the deeper issue of how we handle chat/client packets; just blasting them any time regardless of whether the client is ready.

In this case, I wouldn't expect @kaincenteno to rework the message handling system or invent a whole new delayed afterAfterZoneIn function so the timer is the simplest most reasonable way to get around this one issue for now. :+1:

Reply:
Since we're not in the service of players, there's no rush to merge a PR.

We can do things properly.

Reply:
@Kreidos I didn't see anybody asking him to invent `afterAfterZoneIn`

Reply:
Please mark this PR as on hold, don't really wanna delete it as everything is almost ready :smile_cat: 

I'll add this PR to the issue that already exists of the packet timing issue as @TeoTwawki  explain above :cat: